### PR TITLE
[test] Implement rom_e2e_shutdown_exception_asm

### LINKED ISSUE #14273: [rom-e2e] rom_e2e_shutdown_exception_asm
**Testpoint name:** [rom_e2e_shutdown_exception_asm](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/data/rom_e2e_testplan.hjson?q=rom_e2e_shutdown_exception_asm)

**Contact person:** @alphan

**Description:** Verify that ROM asm exception handler resets the chip.

- Power on with the `CREATOR_SW_CFG_ROM_EXEC_EN` OTP item set to `0`.
  - Execution should halt very early in `_rom_start_boot`.
- Connect the debugger and set a breakpoint at `_asm_exception_handler`.
  - Note: We need to use a debugger for this test since `mtvec` points to C handlers
    when `rom_main()` starts executing.
- Set `pc` to `0x10000000` (start of main SRAM) and execute one machine instruction,
  i.e. `stepi`.
- Verify that execution stops at `_asm_exception_handler` since code execution from SRAM is
  not enabled.
- Continue and verify that the asm exception handler resets the chip by confirming that
  execution halts at `_rom_start_boot`.

### PR DESCRIPTION
**Pull request by dmcardle**
Fixes #14273

Signed-off-by: Dan McArdle <dmcardle@google.com>

### CODE REVIEW

File: sw/device/silicon_creator/rom/e2e/BUILD [hunk 1]
```diff
@@ -1707,6 +1707,63 @@ test_suite(
...
+        ],
+    )
+    for otp_name in OTP_CFGS_EXEC_DISABLED
+]
+
+test_suite(
+    name = "rom_e2e_shutdown_exception_asm",
+    tags = [
+        "cw310",
+        "vivado",
```
Comment (alphan) [inline sw/device/silicon_creator/rom/e2e/BUILD:1762]:
```suggestion
        "manual"
```

File: sw/device/silicon_creator/rom/e2e/BUILD [hunk 2]
```diff
@@ -1707,6 +1707,63 @@ test_suite(
...
+        },
+        rom_bitstream = ":rom_otp_{}_exec_disabled".format(otp_name),
+        rom_kind = "Rom",
+        tags = [
+            "cw310",
+            "exclusive",
+            "vivado",
+        ],
+    )
+    for otp_name in OTP_CFGS_EXEC_DISABLED
```
Comment (alphan) [inline sw/device/silicon_creator/rom/e2e/BUILD:1755]: nit: can you add `maybe_skip_in_ci`?

**Review by alphan — APPROVED** [review]

File: sw/device/silicon_creator/rom/e2e/BUILD (hunk 2 above)
Comment (dmcardle) [reply sw/device/silicon_creator/rom/e2e/BUILD:1755]: Done!

File: sw/device/silicon_creator/rom/e2e/BUILD (hunk 1 above)
Comment (dmcardle) [reply sw/device/silicon_creator/rom/e2e/BUILD]: Done!

File: sw/device/silicon_creator/rom/e2e/BUILD [hunk 3]
```diff
@@ -1707,6 +1707,60 @@ test_suite(
...
+        gdb_script_symlinks = {
+            "//sw/device/silicon_creator/rom:rom_fpga_cw310.elf": "rom.elf",
+        },
+        rom_bitstream = ":rom_otp_{}_exec_disabled".format(otp_name),
+        rom_kind = "Rom",
+        tags = [
+            "cw310",
+            "exclusive",
+            "vivado",
+        ] + maybe_skip_in_ci(structs.to_dict(CONST.LCV)[otp_name.upper()]),
```
Comment (dmcardle) [inline sw/device/silicon_creator/rom/e2e/BUILD:1753]: Forgive me for what I have done.

Comment (drewmacrae) [reply sw/device/silicon_creator/rom/e2e/BUILD:1753]: I think you'll need to add cw310_rom if you want it to run in CI.

**Comment by drewmacrae** [comment]
Re-add the ready to merge tag if you meant for these tests not to run in CI, or if you don't want to add the cw310_rom tag so they match the filters.

**Comment by alphan** [comment]
From "Experimental CW310 ROM Tests":
```
//sw/device/silicon_creator/rom/e2e:shutdown_exception_asm_otp_rma       PASSED in 18.0s
```

**Pull request merged**